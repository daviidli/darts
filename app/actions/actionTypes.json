{
	"SET_SERIAL_PORT": "SET_SERIAL_PORT",
	"SET_WAIT_TIME": "SET_WAIT_TIME",
	"SET_PLAYERS": "SET_PLAYERS",
	"ADD_PLAYER": "ADD_PLAYER",
	"REMOVE_PLAYER": "REMOVE_PLAYER",
	"CHANGE_PLAYER": "CHANGE_PLAYER",
	"SET_MAX_DARTS": "SET_MAX_DARTS",
	"SET_MAX_ROUNDS": "SET_MAX_ROUNDS",
	"SET_CURRENT_PLAYER": "SET_CURRENT_PLAYER",
	"SET_CURRENT_DART": "SET_CURRENT_DART",
	"SET_CURRENT_ROUND": "SET_CURRENT_ROUND",
	"INITIALIZE_ROUNDS": "INITIALIZE_ROUNDS",
	"SET_ROUND": "SET_ROUND",
	"SET_TOTALS": "SET_TOTALS",
	"SET_WINNER": "SET_WINNER",
	"SET_WAITING": "SET_WAITING"
}
