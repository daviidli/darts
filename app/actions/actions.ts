import actions from './actions.json';

export const setSerialPort = (port: string) => ({
	type: actions.SET_SERIAL_PORT,
	port
});

export const test = () => {
	return {};
};
