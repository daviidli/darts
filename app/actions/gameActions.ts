import actions from './actionTypes.json';
import { playerType } from '../reducers/types';

export const setPlayers = (players: playerType[]) => ({
	type: actions.SET_PLAYERS,
	players
});

export const addPlayer = (player: string) => ({
	type: actions.ADD_PLAYER,
	player
});

export const removePlayer = (index: number) => ({
	type: actions.REMOVE_PLAYER,
	index
});

export const changePlayer = (player: string, index: number) => ({
	type: actions.CHANGE_PLAYER,
	player,
	index
});

export const setMaxDarts = (max: number) => ({
	type: actions.SET_MAX_DARTS,
	max
});

export const setMaxRounds = (max: number) => ({
	type: actions.SET_MAX_ROUNDS,
	max
});

export const setCurrentPlayer = (index: number) => ({
	type: actions.SET_CURRENT_PLAYER,
	index
});

export const setCurrentDart = (index: number) => ({
	type: actions.SET_CURRENT_DART,
	index
});

export const setCurrentRound = (index: number) => ({
	type: actions.SET_CURRENT_ROUND,
	index
});

export const initializedRound = (rounds: string[][][]) => ({
	type: actions.INITIALIZE_ROUNDS,
	rounds
});

export const setRound = (
	player: number,
	round: number,
	dart: number,
	roundValue: string
) => ({
	type: actions.SET_ROUND,
	player,
	round,
	dart,
	roundValue
});

export const setTotals = (totals: any[]) => ({
	type: actions.SET_TOTALS,
	totals
});

export const setWinner = (winner: number) => ({
	type: actions.SET_WINNER,
	winner
});

export const setWaiting = (waiting: boolean) => ({
	type: actions.SET_WAITING,
	waiting
});
