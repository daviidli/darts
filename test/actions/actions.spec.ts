import * as actions from '../../app/actions/actions';

describe('actions', () => {
	it('should create setSerialPort action', () => {
		expect(actions.setSerialPort('somePort')).toMatchSnapshot();
	});

	it('should create setWaitTime action', () => {
		expect(actions.setWaitTime(0)).toMatchSnapshot();
	});
});
