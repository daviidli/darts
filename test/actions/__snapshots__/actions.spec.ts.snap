// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions should create setSerialPort action 1`] = `
Object {
  "port": "somePort",
  "type": "SET_SERIAL_PORT",
}
`;

exports[`actions should create setWaitTime action 1`] = `
Object {
  "type": "SET_WAIT_TIME",
  "waitTime": 0,
}
`;
